src/stream/src/executor/source.rs

```diff
@@ -50,12 +55,16 @@ struct SourceReader {
 }
 
 /// `SourceReader` will be turned into this stream type.
-type ReaderStream =
-    Pin<Box<dyn Stream<Item = Either<Result<Message>, Result<StreamChunk>>> + Send>>;
+type ReaderStream = Pin<
+    Box<
+        dyn Stream<Item = Either<Result<Message>, Result<(StreamChunk, HashMap<String, String>)>>>
```

Comment:
Add comments or wrap around `(StreamChunk, HashMap<String, String>)` with a struct to describe what `HashMap<String, String>)` stands for.